### PR TITLE
Stub for proposed API: ExternalUriOpener

### PR DESCRIPTION
**Pull request by alvsan09**
#### What it does

This change enables the use of the `Simple Browser` extension in the context of `vscode` API v1.77.x.

This is accomplished by stubbing missing `vscode` proposed API i.e. `ExternalUriOpener`

#### How to test

1) Remove the existing version of `vscode.simple-browser` (previous builtin version to 1.77.2) from the the `theia` plugins folder
2) Unzip the following `zip` file inside your `plugins` folder (this is the 1.77.2 version of the simple-browser plugin built from sources).
[simple-browser.zip](https://github.com/eclipse-theia/theia/files/11490046/simple-browser.zip)
3) Start `theia` and from the command palette execute the command `Simple Browser: Show`)
    -> on the input box enter the following site: `https://xkcd.com/`
4) Verify that the page open as expected.

NOTE: This browser may not work for many sites, but that's a separate issue to follow up after the true implementation of this stubbed API.

See the following video clip: 
https://github.com/eclipse-theia/theia/assets/76971376/d11e012d-106d-4512-a5e6-729571c896a4

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/plugin/types-impl.ts [hunk 1]
```diff
@@ -229,6 +229,51 @@ export enum SourceControlInputBoxValidationType {
...
+    /**
+     * The opener can open the uri and should be automatically selected over any
+     * default openers, include the built-in one from the editor.
+     *
+     * A preferred opener will be automatically selected if no other preferred openers
+     * are available. If multiple preferred openers are available, then the user
+     * is shown a prompt with all potential openers (not just preferred openers).
+     */
+    Preferred = 3,
+}
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/plugin/types-impl.ts:275]: minor: we can probably remove the documentation given it is just duplication from `packages/plugin/src/theia.proposed.externalUriOpener.d.ts` anyways and simplifies the changes.

File: packages/plugin-ext/src/plugin/plugin-context.ts [hunk 2]
```diff
@@ -573,6 +574,11 @@ export function createAPIFactory(
             },
             get tabGroups(): theia.TabGroups {
                 return tabsExt.tabGroups;
+            },
+
+            // ExternalUriOpener @stubbed
+            registerExternalUriOpener(id: string, opener: theia.ExternalUriOpener, metadata: theia.ExternalUriOpenerMetadata): theia.Disposable {
+                return new Disposable(() => { });
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/plugin/plugin-context.ts:581]: minor - we can use the following instead:

```suggestion
                return Disposable.NULL;
```

**Review by vince-fugnitto — COMMENTED** [review]
LGTM :+1: 
I confirmed that:
- [x] the `vscode.simple-browser` exception is not through regarding `ExternalUriOpener`
- [x] the extension successfully previews the webpage in the browser
- [x] the extension successfully previews the webpage in electron

**Comment by alvsan09** [comment]
Hi @vince-fugnitto, both of your comments were applied and the branch was re-based to the current master,
Thanks !!

**Pull request merged**